tracks/rust/exercises/bob/mentoring.md

```diff
@@ -31,30 +31,31 @@ const CALM: &str = "Calm down, I know what I'm doing!";
 const FINE: &str = "Fine. Be that way!";
 const WHATEVER: &str = "Whatever.";
 
-#[allow(clippy::match_bool)]
 pub fn reply(message: &str) -> &str {
-    let trimmed: &str = message.trim();
-
-    match trimmed.is_empty() {
-        true => FINE,
-        false => {
-            let is_question: bool = trimmed.ends_with("?");
-
-            let contains_alphabetic_characters: bool = trimmed.chars().any(|character| character.is_alphabetic());
-            let is_uppercase: bool = trimmed == trimmed.to_uppercase();
-            let is_yelled: bool = contains_alphabetic_characters && is_uppercase;
-
-            match (is_question, is_yelled) {
-                (true, false) => SURE,
-                (false, true) => WHOA,
-                (true, true) => CALM,
-                (false, false) => WHATEVER,
-            }
+    let message = message.trim();
+
+    if message.is_empty() {
+        FINE
+    } else {
+        let is_question = message.ends_with('?');
+
+        let contains_alphabetic = message.chars().any(char::is_alphabetic);
+        // A message is uppercase if it doesn't contain lowercase.
+        // `.all(char::is_uppercase)` doesn't work because of whitespace/punctuation
+        let is_uppercase = !message.chars().any(char::is_lowercase);
```

Comment:
I just saw this in a student's work:
```rust
let is_uppercase = message.chars().any(char::is_uppercase)
    && !message.chars().any(char::is_lowercase);
```
Should be a bit faster even, since it will short-circuit in more cases than just testing for at least one alphabetic char.

Reply:
You're right. I will change it.

I believe the most efficient (but not the most beautiful) solution is this:

```rust
let message = message.trim_start_matches(|c: char| !c.is_alphabetic());
!message.is_empty() && !message.chars().any(char::is_lowercase)
```